Source/AwsCommonRuntimeKit/mqtt/Mqtt5Packets.swift

```diff
@@ -403,6 +423,65 @@ public class DisconnectPacket {
             self.serverReference = serverReference
             self.userProperties = userProperties
         }
+
+    typealias RawType = aws_mqtt5_packet_disconnect_view
+    func withCStruct<Result>(_ body: (aws_mqtt5_packet_disconnect_view) -> Result) -> Result {
+        var raw_disconnect_view = aws_mqtt5_packet_disconnect_view()
+
+        raw_disconnect_view.reason_code = aws_mqtt5_disconnect_reason_code(UInt32(reasonCode.rawValue))
+
+        let _sessionExpiryInterval = try? sessionExpiryInterval?.secondUInt32() ?? nil
+
+        return withOptionalUnsafePointer(to: _sessionExpiryInterval) { sessionExpiryIntervalPointer in
+
+            if let _sessionExpiryIntervalPointer = sessionExpiryIntervalPointer {
+                raw_disconnect_view.session_expiry_interval_seconds = _sessionExpiryIntervalPointer
+            }
+
+            return withOptionalUserPropertyArray(
+                of: userProperties) { userPropertyPointer in
+
+                if let _userPropertyPointer = userPropertyPointer {
+                    raw_disconnect_view.user_property_count = userProperties!.count
+                    raw_disconnect_view.user_properties =
+                        UnsafePointer<aws_mqtt5_user_property>(_userPropertyPointer)
+                }
+
+                return withOptionalByteCursorPointerFromStrings(
+                    reasonString,
+                    serverReference) { cReasonString, cServerReference in
+                        raw_disconnect_view.reason_string = cReasonString
+                        raw_disconnect_view.server_reference = cServerReference
+                        return body(raw_disconnect_view)
+                    }
+            }
+        }
+    }
+
+    static func convertFromNative(_ from: UnsafePointer<aws_mqtt5_packet_disconnect_view>?) -> DisconnectPacket? {
+        if let from = from {
+            guard let reasonCode = DisconnectReasonCode(rawValue: Int(from.pointee.reason_code.rawValue))
+            else { fatalError("DisconnectPacket from native missing a reason code.") }
+
+            let sessionExpiryInterval = convertOptionalUInt32(from.pointee.session_expiry_interval_seconds)
+            let sessionExpiryIntervalSeconds: TimeInterval? = sessionExpiryInterval.map { TimeInterval($0) }
+            let reasonString = convertAwsByteCursorToOptionalString(from.pointee.reason_string)
```

Comment:
As `from.pointee` is used multiple places, we could simplify it a little. 
```suggestion
            let view = from.pointee
            let reasonString = convertAwsByteCursorToOptionalString(view.reason_string)
```